pkgs/development/node-packages/default-v5.nix

```diff
@@ -0,0 +1,35 @@
+{pkgs, system}:
+
+let
+  nodePackages = import ./composition-v5.nix {
+    inherit pkgs system;
+  };
+in
+nodePackages // {
+  node-inspector = nodePackages.node-inspector.override (oldAttrs: {
+    buildInputs = oldAttrs.buildInputs ++ [ nodePackages.node-pre-gyp ];
+  });
+  
+  phantomjs = nodePackages.phantomjs.override (oldAttrs: {
+    buildInputs = oldAttrs.buildInputs ++ [ pkgs.phantomjs2 ];
+  });
+  
+  webdrvr = nodePackages.webdrvr.override (oldAttrs: {
+    buildInputs = oldAttrs.buildInputs ++ [ pkgs.phantomjs ];
+    
+    preRebuild = ''
+      mkdir $TMPDIR/webdrvr
+      
+      ln -s ${pkgs.fetchurl {
+        url = "https://selenium-release.storage.googleapis.com/2.43/selenium-server-standalone-2.43.1.jar";
+        sha1 = "ef1b5f8ae9c99332f99ba8794988a1d5b974d27b";
+      }} $TMPDIR/webdrvr/selenium-server-standalone-2.43.1.jar
+      ln -s ${pkgs.fetchurl {
+        url = "http://chromedriver.storage.googleapis.com/2.10/chromedriver_linux64.zip";
+        sha1 = "26220f7e43ee3c0d714860db61c4d0ecc9bb3d89";
+      }} $TMPDIR/webdrvr/chromedriver_linux64.zip
+    '';
+    
+    dontNpmInstall = true; # We face an error with underscore not found, but the package will work fine if we ignore this.
```

Comment:
I agree that this is very ugly -- however, what I'm basically doing here is following the conventions the stdenv generic builder uses, e.g. for `stdenv.mkDerivation {}` you can set do* and dont* flags to enable or disable certain steps, e.g.:

```nix
stdenv.mkDerivation {
  ...
  doCheck = true;
  dontStrip = true;
}
```

so it's as nice (or as ugly!) the generic builder is.

Also, in the future I'm planning to create a better structured abstraction function that (for example) uses the generic builder's phases and so on.